### PR TITLE
XDR-5434: Add NPM tests

### PR DESCRIPTION
**Pull request by GerardSetho**
This will add optional tests to the validate.yaml workflow.

```yaml
jobs:
  ci:
    name: Validation
    uses: cncsc/actions/.github/workflows/validation.yaml@main
    with:
      runPreCommit: false # optional
      runDefaultLinters: false # optional
      npmTestScripts: test e2e # optional
```

This will run the below commands:
- `npm run test`
- `npm run e2e`

### CODE REVIEW

File: README.md
Comment (straineanu) [inline README.md:25]: Thanks for updating the readme

File: .github/workflows/validation.yaml
Comment (straineanu) [inline .github/workflows/validation.yaml:27]: `required: false` makes this a non breaking change, right? Should make it so that any repository currently referencing `@main` in their github workflow won't break once this PR is merged to main.

File: .gitignore
Comment (straineanu) [inline .gitignore:9]: Not related to npm tests or validation. I don't think Armor has strict standards around this yet. I think a bit of process overhead is worth having the PRs more focused. Don't need to move this now, I'm just using the soapbox to preach a bit.

**Review by straineanu — APPROVED** [review]
Still needs a codeowner approval for actual merging. Looks reasonable to me.

File: .gitignore
Comment (GerardSetho) [reply .gitignore:9]: Good point. In the future, I'll be sure to keep the PRs as focused as possible.

File: .github/workflows/validation.yaml
Comment (GerardSetho) [reply .github/workflows/validation.yaml:27]: Yes correct.

File: .github/workflows/validation.yaml
Comment (lukiffer) [inline .github/workflows/validation.yaml:103]:
```suggestion
  # -------------------------------------------------------------------------------------------------------------------
  # RUN PRE-COMMIT VALIDATION
  # -------------------------------------------------------------------------------------------------------------------
  run-tests:
```

Would add a comment header here just to keep it consistent.

File: .github/workflows/validation.yaml
Comment (lukiffer) [inline .github/workflows/validation.yaml:122]: I think there are advantages and disadvantages to this bash approach. It's a bit arbitrary though given that the list of tasks supplied in `npmTestScripts` needn't actually be test scripts (i.e. you could supply a value of `deploy` and end up running `npm run deploy` in a "test" job.

I think one of the disadvantages is that if the first job in sequence fails, you don't get test telemetry from subsequent jobs. In that regard, it might be better to run this as a matrix job.

The last piece of commentary would be that we typically haven't been this specific in past (where npm jobs needed to be specifically called). The `lint` script for example in some projects encapsulates multiple "sub-jobs":

```json
{
  "scripts": {
    "lint:ts": "eslint ...",
    "lint:docker": "hadolint ...",
    "lint": "npm run lint:ts && npm run lint:docker"
  }
}
```

This approach allows the `lint` script to be run by convention and the specifics of what gets linted is left to the project (via `package.json` configuration). This is more a vestige of our time using Makefiles, and might not be the best approach anymore – but if we do change course, we should probably work to have consistent interfaces across linters, tests, and other scripts that may have subcomponents.

**Pull request closed without merging**
